src/test/java/com/exasol/matcher/ResultSetMatcher.java

```diff
@@ -0,0 +1,119 @@
+    private final ResultSet expectedResultSet;
+
+    /**
+     * Creates a new instance of {@link ResultSetMatcher}.
+     * 
+     * @param expectedResultSet expected Result Set
+     */
+    public ResultSetMatcher(final ResultSet expectedResultSet) {
+        this.expectedResultSet = expectedResultSet;
+    }
+
+    /**
+     * Compares against a result set.
+     * 
+     * @param expectedResultSet expected result set
+     * @return an instance of {@link ResultSetMatcher}
+     */
+    public static ResultSetMatcher matchesResultSet(final ResultSet expectedResultSet) {
+        return new ResultSetMatcher(expectedResultSet);
+    }
+
+    @Override
+    protected boolean matchesSafely(final ResultSet actualResultSet) {
+        try {
+            return assertEqualResultSets(actualResultSet);
+        } catch (final SQLException exception) {
+            throw new AssertionError(
+                    "Assertion failed due to an unexpected SQL exception. Cause: " + exception.getSQLState(),
+                    exception);
+        }
+    }
+
+    @Override
+    public void describeTo(final Description description) {
+        description.appendValue(this.expectedResultSet);
+    }
+
+    private boolean assertEqualResultSets(final ResultSet actualResultSet) throws SQLException {
+        final int expectedColumnCount = this.expectedResultSet.getMetaData().getColumnCount();
+        final int actualColumnCount = actualResultSet.getMetaData().getColumnCount();
+        if (expectedColumnCount != actualColumnCount) {
+            return false;
+        }
+        boolean expectedNext;
+        do {
+            expectedNext = this.expectedResultSet.next();
+            if (expectedNext != actualResultSet.next()) {
+                return false;
+            }
+            if (this.expectedResultSet.isLast() != actualResultSet.isLast()) {
+                return false;
+            }
+            if (expectedNext) {
+                if (!rowMatches(actualResultSet, expectedColumnCount)) {
+                    return false;
+                }
+            }
+        } while (expectedNext);
+        return true;
+    }
+
+    private boolean rowMatches(ResultSet actualResultSet, int expectedColumnCount) throws SQLException {
+        for (int column = 1; column <= expectedColumnCount; ++column) {
+            final int resultSetTypeExpected = this.expectedResultSet.getMetaData().getColumnType(column);
+            final int resultSetTypeActual = actualResultSet.getMetaData().getColumnType(column);
+            if (resultSetTypeExpected == resultSetTypeActual) {
+                if (!valueMatches(actualResultSet, column, resultSetTypeExpected)) {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean valueMatches(ResultSet actualResultSet, int column, int resultSetTypeExpected) throws SQLException {
```

Comment:
Make all parameters final (also for other methods).

Reply:
Rename method to `doesValueMatch()`.
Rename first parameter to `actualRow`.

Reply:
done